src/app/expansion-panel/expansion-panel.component.ts

```diff
@@ -8,12 +9,23 @@ import { ChangeDetectionStrategy, Component, EventEmitter, Output } from '@angul
   templateUrl: './expansion-panel.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ExpansionPanelComponent {
+export class ExpansionPanelComponent implements OnInit {
+  @Input() questionPanel?: boolean = false;
   @Output() expandedChanged = new EventEmitter<void>();
-  public expanded: boolean = true;
+  public expanded: boolean = defaultConfig.startsExpanded;
```

Comment:
```suggestion
  public config: ExpansionPanelConfig = defaultConfig;
```
Combine all of the public parts of the config.

Reply:
I've also seen the non-null assertion used in that way, one of those things that's technically unsafe but you'd find out pretty quickly if it was ever changed so it isn't set.

Another option would be to rename startsExpanded to just expanded and use/toggle the value that's inside the config object. I don't know if either feels better to you?

Reply:
Cool, that's fine by me to go for the non-null assertion version.

The point about local state and not modifying things that are passed in feels valid. I think in this instance it would be okay to use the expanded value as the ... spread operator creates a copy of the bool property but if we had some more complex nested objects then you would potentially be modifying external objects.

Reply:
Ah yes, I hadn't considered that we were copying into a new object. I still feel like the config object that gets passed in should be immutable but the internal config would need a mutable 'expanded' field.

Reply:
I think overall I prefer keeping the state and the config separate and if we added more state in the future it wouldn't necessarily belong in the config object so we could end up with state stored in slightly different ways in the component. Happy to defer though if you prefer having it as a config field.